### PR TITLE
CreateOrganization View

### PR DESCRIPTION
**Pull request by sanyavertolet**
This issue covers almost all the #562 

### What's done:
* Added button to create new Organization
* Added new View for the creation of the organization
* Created backend endpoint that will create organization

(#562)

### CODE REVIEW

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt [hunk 1]
```diff
@@ -28,4 +31,32 @@ internal class OrganizationController(private val organizationService: Organizat
...
+        val ownerId = (authentication.details as AuthenticationDetails).id
+        val (organizationId, organizationStatus) = organizationService.saveOrganization(
+            newOrganization.apply {
+                this.ownerId = ownerId
+                this.dateCreated = LocalDateTime.now()
+            }
+        )
+
+        if (organizationStatus == OrganizationSaveStatus.EXIST) {
+            logger.warn("Organization with id = $organizationId already exists")
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:51]: Will any data get overwritten in this case?

File: save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/OrganizationSaveStatus.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package org.cqfn.save.domain
+
+import kotlinx.serialization.Serializable
+
+/**
+ * Enum of organization save status
+ * @property message
+ */
+@Serializable
+@Suppress("CUSTOM_GETTERS_SETTERS")
```
Comment (petertrr) [inline save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/OrganizationSaveStatus.kt:10]: Why is this needed?

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt [hunk 3]
```diff
@@ -10,6 +14,27 @@ import org.springframework.stereotype.Service
...
+     * Store [organization] in the database
+     *
+     * @param organization an [Organization] to store
+     * @return organization's id, should never return null
+     */
+    @Suppress("UnsafeCallOnNullableType")
+    fun saveOrganization(organization: Organization): Pair<Long, OrganizationSaveStatus> {
+        val exampleMatcher = ExampleMatcher.matchingAll()
+            .withMatcher("name", ExampleMatcher.GenericPropertyMatchers.exact())
+            .withMatcher("ownerId", ExampleMatcher.GenericPropertyMatchers.exact())
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt:27]: I recognize copy-paste from `ProjectService`, but doesn't `matchingAll` already mean all fields match `exact`? I don't remember, why we use it like this there, but I think it's redundant.

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt:27]: Oh yes, there was a copy-paste from `ProjectService`. Now it is something like that:
```
val exampleMatcher = ExampleMatcher.matching()
            .withMatcher("name", ExampleMatcher.GenericPropertyMatchers.exact())
```
But I cannot see what is the difference between `matching` and `matchingAll` cause according to [`docs.spring.io`](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/domain/ExampleMatcher.html) they do exactly the same.

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt:27]: Actually this is what source code looks like:
```
static ExampleMatcher matching() {
	return matchingAll();
}
```

File: save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/OrganizationSaveStatus.kt (hunk 2 above)
Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/OrganizationSaveStatus.kt:10]: Copy-paste problem, deleted it.

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:51]: No, it will not because if there is an existing organization that conflicts the new one, an existing one will be returned from `saveOrganization`.

Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:51]: Right, we are checking if it exists inside the method. Maybe, then rename it to something like `getOrSaveOrganization`?

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt (hunk 3 above)
Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt]: I mean, can we simply drop two lines with specific matchers and only leave `matchingAll`? I believe, result will be the same

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt]: The idea is to match only name because either we get one organization per user (`matchingAny`) or many organizations with same name (`matching`/`matchingAll`)

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:51]: Maybe but we need to rename same method for `ProjectController`. I guess it would be nice to keep same format for same things in different classes.

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt (hunk 3 above)
Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt]: Then we can simply use `findByName`, because example matcher is better suited for different case - when you would like to match against multiple fields

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt]: Oh ok, will fix it soon

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt]: Done

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt [hunk 4]
```diff
@@ -23,14 +23,14 @@ internal class OrganizationController(private val organizationService: Organizat
...
-            organizationService.findByName(name)
+            organizationService.findByName(name) ?: throw NoSuchElementException("Organization with name [$name] was not found.")
 
     /**
      * @param owner owner name
      * @return a image
      */
     @GetMapping("/avatar")
-    fun avatar(@RequestParam owner: String): ImageInfo? = organizationService.findByName(owner).avatar.let { ImageInfo(it) }
+    fun avatar(@RequestParam owner: String): ImageInfo? = organizationService.findByName(owner)?.avatar.let { ImageInfo(it) }
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:33]: Should it be `?.let`? Otherwise function never returns null, only null wrapped into `ImageInfo`. If that's intended, we can change returned type

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt [hunk 5]
```diff
@@ -51,9 +44,9 @@ class OrganizationService(private val organizationRepository: OrganizationReposi
      * @param relativePath
      */
     fun saveAvatar(name: String, relativePath: String) {
-        val organization = organizationRepository.findByName(name).apply {
+        val organization = organizationRepository.findByName(name)?.apply {
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt:47]: Please handle missing organization here (maybe, with an exception). Now you silently ignore missing organization and don't indicate that no avatar has actually been saved.

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt [hunk 6]
```diff
@@ -20,12 +23,40 @@ internal class OrganizationController(private val organizationService: Organizat
...
+        val ownerId = (authentication.details as AuthenticationDetails).id
+        val (organizationId, organizationStatus) = organizationService.getOrSaveOrganization(
+            newOrganization.apply {
+                this.ownerId = ownerId
+                this.dateCreated = LocalDateTime.now()
+            }
+        )
+
+        if (organizationStatus == OrganizationSaveStatus.EXIST) {
+            logger.warn("Organization with id = $organizationId already exists")
```
Comment (petertrr) [inline save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:51]:
```suggestion
            logger.info("Attempt to save an organization with id = $organizationI, but it already exists")
```
I think it shouldn't be a warning, because it is a valid and properly handled case. Otherwise it'll just pollute logs.

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:51]: Ok, agree

File: save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt (hunk 5 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/service/OrganizationService.kt:47]: Yes, true. Now throwing exception.

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt (hunk 4 above)
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt:33]: Ok, changed return type.

**Review by petertrr — APPROVED** [review]
